kfs-core/src/main/java/edu/arizona/kfs/fp/batch/service/impl/ProcurementCardCreateDocumentServiceImpl.java

```diff
@@ -359,53 +386,148 @@ protected ProcurementCardDefault retrieveProcurementCardDefault(String creditCar
+                } catch (WorkflowException e) {
+                    // If an individual document has a problem with being auto approved, it should not stop the job from approving the rest.
+                    LOG.error("Error auto approving document # " + pcardDocument.getDocumentNumber() + " " + e.getMessage(), e);
+                }
+            }
+        }
 
-        // get number of days and type for auto approve
-        int autoApproveNumberDays = Integer
-                .parseInt(super.parameterService.getParameterValueAsString(ProcurementCardAutoApproveDocumentsStep.class,
-                        ProcurementCardDocumentRuleConstants.AUTO_APPROVE_NUMBER_OF_DAYS));
+        if (autoApproveErrorStr.length() > 0) {
+            outputErrorFile(autoApproveErrorStr.toString());
+        }
 
-        Timestamp currentDate = dateTimeService.getCurrentTimestamp();
-        for (ProcurementCardDocument pcardDocument: pcardDocumentList) {
-            try {
+        return true;
 
-                // prevent PCard documents from auto approving if they have capital asset info to collect
-                if(capitalAssetBuilderModuleService.hasCapitalAssetObjectSubType(pcardDocument)) {
-                    continue;
-                }
+    }
 
-                // if number of days in route is passed the allowed number, call doc service for super user approve
-                Timestamp docCreateDate = new Timestamp( pcardDocument.getDocumentHeader().getWorkflowDocument().getDateCreated().getMillis() );
-                if (KfsDateUtils.getDifferenceInDays(docCreateDate, currentDate) > autoApproveNumberDays) {
-                    // update document description to reflect the auto approval
-                    if (pcardDocument.getDocumentHeader().getDocumentDescription().startsWith("FY ")) {
-                        pcardDocument.getDocumentHeader().setDocumentDescription(pcardDocument.getDocumentHeader().getDocumentDescription().substring(0, 7) + " Auto Approve " + dateTimeService.toDateTimeString(currentDate));
-                    } else {
-                        pcardDocument.getDocumentHeader().setDocumentDescription("Auto Approved On " + dateTimeService.toDateTimeString(currentDate) + ".");
-                    }
-                    LOG.info("Auto approving document # " + pcardDocument.getDocumentHeader().getDocumentNumber());
-                    pcardDocument.setAutoApprovedIndicator(true);
-                    getDocumentService().superUserApproveDocument(pcardDocument, "");
+    /**
+     * Returns true if the document contains an accounting line that is restricted from being auto-approved.
+     * 
+     * @param pcardDocument
+     * @param accountValues
+     * @param objectSubTypeValues
+     * @return
+     */
+    private boolean checkAutoApproveAccountingLineRestriction(ProcurementCardDocument pcardDocument, Collection<String> accountValues, Collection<String> objectSubTypeValues) {
+        @SuppressWarnings("unchecked")
```

Comment:
This was done to suppress warnings that were showing up in Eclipse. pcardDocument.getTargetAccountingLines() returns only a List in the delivered ProcurementCardDocument. Last I was aware, it was SOP to suppress warnings that were caused by delivered classes/interfaces. If this has changed, I can certainly remove it.